DeviceAdapters/Zaber/Zaber.cpp

```diff
@@ -118,281 +124,145 @@ ZaberBase::~ZaberBase()
+int ZaberBase::GetSettings(long device, long axis, std::string setting, std::vector<double>& data)
 {
-	core_->LogMessage(device_, "ZaberBase::GetSetting(double)\n", true);
-
-	ostringstream cmd;
-	cmd << cmdPrefix_ << device << " " << axis << " get " << setting;
-	vector<string> resp;
+	core_->LogMessage(device_, "ZaberBase::GetSettings()\n", true);
 
-	int ret = QueryCommand(cmd.str().c_str(), resp);
-	if (ret != DEVICE_OK) 
+	data.clear();
+	zml::Response resp;
+	int ret = Command(device, axis, "get " + setting, resp);
+	if (ret != DEVICE_OK)
 	{
+		if (ret == DEVICE_UNSUPPORTED_COMMAND) {
+			return DEVICE_OK;
+		}
 		return ret;
 	}
 
-	// extract data
-	string dataString = resp[5];
-	stringstream(dataString) >> data;
-	return DEVICE_OK;
-}
-
-
-int ZaberBase::SetSetting(long device, long axis, string setting, long data) const
-{
-	core_->LogMessage(device_, "ZaberBase::SetSetting(long)\n", true);
-
-	ostringstream cmd; 
-	cmd << cmdPrefix_ << device << " " << axis << " set " << setting << " " << data;
-	vector<string> resp;
+	std::vector<string> strData;
+	CDeviceUtils::Tokenize(resp.getData(), strData, " ");
+	for (const auto& token : strData) {
+		if (token == "NA") {
+			data.push_back(NAN);
+			continue;
+		}
 
-	int ret = QueryCommand(cmd.str().c_str(), resp);
-	if (ret != DEVICE_OK)
-	{
-		return ERR_SETTING_FAILED;
+		double numData;
+		stringstream(token) >> numData;
+		data.push_back(numData);
 	}
 
 	return DEVICE_OK;
 }
 
-
-int ZaberBase::SetSetting(long device, long axis, string setting, double data, int decimalPlaces) const
+int ZaberBase::SetSetting(long device, long axis, string setting, double data, int decimalPlaces)
 {
 	core_->LogMessage(device_, "ZaberBase::SetSetting(double)\n", true);
 
-	ostringstream cmd; 
-	cmd.precision(decimalPlaces);
-	cmd << cmdPrefix_ << device << " " << axis << " set " << setting << " " << fixed << data;
-	vector<string> resp;
+	ostringstream cmd;
+	cmd.precision(decimalPlaces > 0 ? decimalPlaces : 0);
+	cmd << "set " << setting << " " << fixed << data;
+	zml::Response resp;
 
-	int ret = QueryCommand(cmd.str().c_str(), resp);
-	if (ret != DEVICE_OK)
+	int ret = Command(device, axis, cmd.str(), resp);
+	if (ret == ERR_COMMAND_REJECTED)
 	{
 		return ERR_SETTING_FAILED;
 	}
-
-	return DEVICE_OK;
+	return ret;
 }
 
 
-bool ZaberBase::IsBusy(long device) const
+bool ZaberBase::IsBusy(long device)
 {
 	core_->LogMessage(device_, "ZaberBase::IsBusy\n", true);
 
-	ostringstream cmd;
-	cmd << cmdPrefix_ << device;
-	vector<string> resp;
+	zml::Response resp;
 
-	int ret = QueryCommand(cmd.str().c_str(), resp);
+	int ret = Command(device, 0, "", resp);
 	if (ret != DEVICE_OK)
 	{
 		ostringstream os;
-		os << "SendSerialCommand failed in ZaberBase::IsBusy, error code: " << ret;
+		os << "ZaberBase::IsBusy failed, error code: " << ret;
 		core_->LogMessage(device_, os.str().c_str(), false);
 		return false;
 	}
 
-	return (resp[3] == ("BUSY"));
+	return resp.getStatus() != "IDLE";

```

Comment:
```suggestion
	return resp.getStatus() == "BUSY";

```

There probably won't ever be a third value, but I think it's more future-proof to use the value that explicitly means busy.